### PR TITLE
Add KuzuPropertyGraphStore

### PR DESCRIPTION
**Pull request by prrao87**
# Description

This PR adds support for `KuzuPropertyGraphStore` that allows users to leverage a [Kùzu](https://kuzudb.com/) graph database backend.

Kùzu implements a structured property graph model, which means that there are node and relationship tables, whose names and overall structure are decided beforehand via the `validation_schema`. Because there are two "modes" of operation in `PropertyGraphIndex` when using the `SchemaLLMPathExtractor`, the example notebook covers both these modes of operation, which could prove useful when the user is experimenting.

I presume that the example notebook will be rendered on the website for reference? If there's any other place to update so that it shows up alongside other graph DB vendors, let me know. Thanks!

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/graph_stores/llama-index-graph-stores-kuzu/pyproject.toml [hunk 1]
```diff
@@ -27,11 +27,11 @@ exclude = ["**/BUILD"]
 license = "MIT"
 name = "llama-index-graph-stores-kuzu"
 readme = "README.md"
-version = "0.2.0"
+version = "0.6.0"
```
Comment (logan-markewich) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-kuzu/pyproject.toml:30]: lol 2 to 6? Sure, why not :) 

Lets also update the `[tool.llamahub.class_authors]` to add the new class

Comment (logan-markewich) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-kuzu/pyproject.toml:30]: (ah, the version matches the kuzu version)

**Review by logan-markewich — COMMENTED** [review]
This looks fine to me, one small comment

File: llama-index-integrations/graph_stores/llama-index-graph-stores-kuzu/pyproject.toml (hunk 1 above)
Comment (prrao87) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-kuzu/pyproject.toml:30]: Actually, I'm not sure what the graph store's version should be 😅.. Should I just change it back to 0.3.0? I don't see us bumping the version for every kuzu version.

Comment (prrao87) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-kuzu/pyproject.toml]: Done! Updated 👍🏽

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**